purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -4474,8 +4473,10 @@ class PurchasesTest {
         offeringIdentifier: String?,
         subscriptionOptionId: String? = this.subscriptionOptionId
     ): ReceiptInfo {
+        val productId = storeProduct.purchasingData.productId
```

Comment:
Tests were failing here because this ended up mocking the `ReceiptInfo` with the a product id with the base plan in it. This fixes that using the subscription/parent id in `purchasingData`.

This feels _less_ gross because its a test but still meh